### PR TITLE
feat(curriculum): add favourite icon toggler lab

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+```js
+const id = document.querySelectorAll('ul li span.favorite-icon');
+assert.equal(id[2]?.id, 'favoriteIcon3');
+```
+
+always passing >> Initially, the `span` elements should contain the code `&#9825;` to represent an empty heart.
+
+```js
+const inputs = document.querySelectorAll('ul li span.favorite-icon');
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md:79]:
```suggestion
const inputs = document.querySelectorAll('ul li span.favorite-icon');
assert(inputs.length)
```
When `inputs` is empty the for..of loop does not run. Therefore the test does not fail.

File: curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+id: 66bf6bacf178eac7b96d4f5e
+title: Build a Favourite Icon Toggler
+challengeType: 14
+dashedName: build-a-favourite-icon-toggler
+demoType: onClick
+---
+
+# --description--
+
+In this lab you will use javascript click events to toggle the appearance of a favorite icon. When the heart icon is clicked, the appearance of the heart changes from empty to filled, and vice versa.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md:11]:
```suggestion
In this lab you will use JavaScript click events to toggle the appearance of a favorite icon. When the heart icon is clicked, the appearance of the heart changes from empty to filled, and vice versa.
```

**Review by jdwilkin4 — COMMENTED** [review]
It also looks like some of the tests are failing too

**Review by jdwilkin4 — APPROVED** [review]

**Review by moT01 — CHANGES_REQUESTED** [review]
This is a fun one @zairahira 👍 

I left a few suggestions below. Additionally, I think you should change all the "favourite"'s, to "favorite"'s - including for the filenames.

Lastly, you could add one more user story and test for something like "Your `filled` class should change the `color` of the content with that class." Or "...should change it to red..."

File: curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**.
+
+**User Stories:**
+
+1. You should have an unordered list with three items.
+2. The unordered list should have the class `item-list`.
+3. The three list items should contain the item name followed by a `span` element with the class `favorite-icon` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3`, respectively.
+4. The `span` element should contain the code `&#9825;` initially to represent an empty heart.
+5. When the `span` element containing the heart is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md:21]: How about this:

5. When a `span` element containing a heart is clicked, you should add the `filled` class to the clicked `span` if it's not already present, and remove it if it is.

6. When a `span` element containing a heart is clicked, the heart symbol should toggle between `&#9825;` (empty heart) and `&#10084;` (filled heart), depending on its current state.

File: curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+In this lab you will use JavaScript click events to toggle the appearance of a favorite icon. When the heart icon is clicked, the appearance of the heart changes from empty to filled, and vice versa.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**.
+
+**User Stories:**
+
+1. You should have an unordered list with three items.
+2. The unordered list should have the class `item-list`.
+3. The three list items should contain the item name followed by a `span` element with the class `favorite-icon` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3`, respectively.
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md]: I would agree, those IDs aren't being used anywhere.
I have removed them from the instruction and the tests.

File: curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (moT01) [reply curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md]: Yea, that would work better I think.

[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md]: @moT01 
I have updated these 2 stories.

Also, the `filled` class is just being used in script logic to identify whether the icon is filled or not. It's not adding a colour or any other style. The icon `&#10084` is already a filled black heart. So, it shouldn't need a separate user story.

File: curriculum/challenges/english/25-front-end-development/lab-favorite-icon-toggler/66bf6bacf178eac7b96d4f5e.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
+---
+id: 66bf6bacf178eac7b96d4f5e
+title: Build a Favourite Icon Toggler
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-favorite-icon-toggler/66bf6bacf178eac7b96d4f5e.md:3]:
```suggestion
title: Build a Favorite Icon Toggler
```

File: curriculum/challenges/_meta/lab-favorite-icon-toggler/meta.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Build a Favourite Icon Toggler",
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "blockType": "lab",
+  "dashedName": "lab-favorite-icon-toggler",
+  "order": 191,
+  "superBlock": "front-end-development",
+  "challengeOrder": [{ "id": "66bf6bacf178eac7b96d4f5e", "title": "Build a Favourite Icon Toggler" }],
```
Comment (moT01) [inline curriculum/challenges/_meta/lab-favorite-icon-toggler/meta.json:9]:
```suggestion
  "challengeOrder": [{ "id": "66bf6bacf178eac7b96d4f5e", "title": "Build a Favorite Icon Toggler" }],
```

File: client/i18n/locales/english/intro.json [hunk 7]
```diff
@@ -2021,6 +2021,12 @@
       "gibb": { "title": "189", "intro": [] },
       "whrm": { "title": "190", "intro": [] },
       "fhxc": { "title": "191", "intro": [] },
+      "lab-favorite-icon-toggler": {
+        "title": "Build a Favourite Icon Toggler",
```
Comment (moT01) [inline client/i18n/locales/english/intro.json:2025]:
```suggestion
        "title": "Build a Favorite Icon Toggler",
```

File: client/src/pages/learn/front-end-development/lab-favorite-icon-toggler/index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+title: Introduction to the Build a Favourite Icon Toggler
```
Comment (moT01) [inline client/src/pages/learn/front-end-development/lab-favorite-icon-toggler/index.md:2]:
```suggestion
title: Introduction to the Build a Favorite Icon Toggler
```

File: curriculum/challenges/_meta/lab-favorite-icon-toggler/meta.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Build a Favourite Icon Toggler",
```
Comment (moT01) [inline curriculum/challenges/_meta/lab-favorite-icon-toggler/meta.json:2]:
```suggestion
  "name": "Build a Favorite Icon Toggler",
```

**Review by moT01 — CHANGES_REQUESTED** [review]
noticed a couple more

Also, looks like there's some conflicts. And it looks like the order on the sheet changed to 190 - so we need to change that in the meta.json file and where the info is in the intro.json file if needed.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**